### PR TITLE
(Feat) server and local database setup

### PR DESCRIPTION
**Pull request by kkinaman**
COMPLETED: 
- Node express server set up (server/index.js)
- Npm dependencies set up (package.json)
- Gruntfile written, basic tasks defined (Gruntfile.js) and npm scripts written (package.json)
- Linting set up
- MongoDB set up (server/config/db-config.js)
- Database schemas defined (server/config/users/userModel.js and server/config/memories/memoryModel.js)
- Routes set up for api calls (server/config/routes.js)
- Methods written for sending POST requests to /api/login and /api/signup (server/config/users/userController.js)
- Tests written for logging in and signing up (test/ServerSpec.js)

NEXT STEPS:
- specify where client files are located for static files (index.js)
- specify client files for linting (Gruntfile.js)
- host a mongodb instance on heroku and start using that

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -0,0 +1,14 @@
+var express = require('express');
+
+var app = express();
+
+
+// app.use(express.static(__dirname + '/client'/* TODO: insert/change client folder path here */));
+
+var port = 3000;
```
Comment (ghost) [inline index.js]: We can use process.env.PORT || 3000 if we decide on Heroku deployment.

File: server/config/memories/memoryModel.js [hunk 2]
```diff
@@ -0,0 +1,12 @@
+var mongoose = require('mongoose');
+var Schema = mongoose.Schema;
+
+var memorySchema = new Schema({
+  title: String,
+  filePath: String,
+  createdAt: Date,
+  /* TODO: How will we store this data? JSON? stringified? tags? */
+  analyses: []
```
Comment (ghost) [inline server/config/memories/memoryModel.js:9]: Hard to say before using the APIs :) but I imagine a decision we'll need to make is either one schema that's broad enough to represent all analyses or a schema per analysis?

File: server/config/users/userModel.js [hunk 3]
```diff
@@ -0,0 +1,14 @@
+var mongoose = require('mongoose');
+var Schema = mongoose.Schema;
+
+var userSchema = new Schema({
+  username: String,
```
Comment (ghost) [inline server/config/users/userModel.js:5]: Should we enforce uniqueness on the schema side as well?

File: server/config/users/userController.js [hunk 4]
```diff
@@ -1,14 +1,24 @@
...
-    res.status(201).send(user);
-  });
+  User.findOne({username: req.body.username})
+    .then(function(user) {
+      if (!user) {
+        User.create({
+          username: req.body.username,
+          password: req.body.password
+        }).then(function(user) {
+          res.status(201).send(user);
```
Comment (ghost) [inline server/config/users/userController.js:22]: Does it matter (from a security perspective) if we send back the password in the response?

File: test/ServerSpec.js [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+    var signUpOptions = {
+      method: 'POST',
+      url: 'http://localhost:3000/api/signup',
+      json: {
+        username: 'Phillip',
+        password: 'Phillip'
+      }
+    };
+
+    it('Should add a new user to the database', function(done) {
```
Comment (ghost) [inline test/ServerSpec.js:23]: Yay for tests! Should we rename this to "Should return user info in response body for successful signups" ? Does it make sense to check that we actually inserted documents into the Mongo database?

File: test/ServerSpec.js [hunk 6]
```diff
@@ -24,6 +24,38 @@ describe('', function() {
...
+    var signUpOptions = {
+      method: 'POST',
+      url: 'http://localhost:3000/api/login',
+      json: {
+        username: 'Phillip',
+        password: 'Phillip'
+      }
+    };
+
+    it('Should log in an existing user to the database', function(done) {
```
Comment (ghost) [inline test/ServerSpec.js:67]: "Should log in existing user" ?

File: server/config/users/userController.js [hunk 7]
```diff
@@ -1,9 +1,14 @@
-
+var User = require('./userModel');
 
 exports.login = function(req, res) {
 
 };
 
 exports.signup = function(req, res) {
-
+  User.create({
```
Comment (ghost) [inline server/config/users/userController.js]: Shall we table the password encryption for a pre-release issue or just decide not to bother with it at all?

**Review by ghost — APPROVED** [review]
Nice job!

File: server/config/users/userController.js (hunk 4 above)
Comment (kkinaman) [reply server/config/users/userController.js:22]: Will need to figure out where authentication will take place (we will need access to the password to authenticate)

**Pull request merged**